lib/psych/visitors/yaml_tree.rb

```diff
@@ -15,30 +15,29 @@ module Visitors
     class YAMLTree < Psych::Visitors::Visitor
       class Registrar # :nodoc:
         def initialize
-          @obj_to_id   = {}
-          @obj_to_node = {}
+          @obj_to_id   = {}.compare_by_identity
+          @obj_to_node = {}.compare_by_identity
           @targets     = []
```

Comment:
@tenderlove Looking again... is `@targets` just completely unused? Only ever shoved into, never read.

Reply:
Removed in #665